velox/exec/Driver.cpp

```diff
@@ -1090,6 +1023,68 @@ folly::dynamic Driver::toJson() const {
   return obj;
 }
 
+template <typename Func>
+auto Driver::withDeltaCpuWallTimer(
+    Operator* op,
+    TimingMemberPtr timingMember,
+    Func&& timedFunction) -> decltype(timedFunction()) {
+  // If 'trackOperatorCpuUsage_' is true, create and initialize the timer object
+  // to track cpu and wall time of the timedFunction.
+  auto timer = trackOperatorCpuUsage_
+      // The delta CpuWallTiming object would be recorded to the corresponding
+      // timingMember upon destruction of the timer when withDeltaCpuWallTimer
+      // ends.
+      ? std::make_unique<
+            DeltaCpuWallTimer<std::function<void(const CpuWallTiming&)>>>(
```

Comment:
`std::function` would be virtual and do potential heap allocation as well.  Use this instead:
```c++
auto f = [op, timingMember, this](const CpuWallTiming& elapsedTime) {
    auto elapsedSelfTime = processLazyTiming(*op, elapsedTime);
    op->stats().withWLock([&](auto& lockedStats) {
    (lockedStats.*timingMember).add(elapsedSelfTime);
  });
};
DeltaCpuWallTimer<decltype(f)> timer(f);
```